lib/perl/Genome/Model/DeNovoAssembly/Command/ExportAsGscProject.pm

```diff
@@ -0,0 +1,176 @@
+package Genome::Model::DeNovoAssembly::Command::ExportAsGscProject;
+
+use strict;
+use warnings;
+
+use File::Basename;
+use Params::Validate qw/ :types validate_pos /;
+
+class Genome::Model::DeNovoAssembly::Command::ExportAsGscProject {
+    is => 'Command::V2',
+    has => {
+        directory => {
+            is => 'Text',
+            doc => 'Directory to put the project.',
+        },
+        project => {
+            is => 'Genome::Project',
+            doc => 'Work Order',
+        },
+    },
+};
+
+sub __errors__ {
+    my $self = shift;
+
+    my @errors = $self->SUPER::__errors__;
+    return @errors if @errors;
+
+    if ( not -d $self->directory ) {
+        return UR::Object::Tag->create(
+            type => 'invalid',
+            properties => [qw/ directory /],
+            desc => 'Directory does not exist: '.$self->directory,
+        );
+    }
+
+    return;
+}
+
+my %supported_assemblers = (
+    'velvet one-button' => {
+        subdirs => [qw/ edit_dir chromat_dir phd_dir/],
+        edit_dir => 'edit_dir',
+    },
+    'newbler de-novo-assemble' => {
+        subdirs => [qw/ edit_dir chromat_dir phd_dir phdball_dir /],
+        edit_dir => File::Spec->join('consed', 'edit_dir'),
+    },
+);
+sub supported_assemblers { keys %supported_assemblers }
+sub is_model_supported {
+    my ($self, $model) = validate_pos(@_, {isa => __PACKAGE__}, {type => OBJECT});
+
+    my $assembler = $model->processing_profile->assembler_name;
+    return if ! List::MoreUtils::any { $assembler eq $_ } supported_assemblers();
```

Comment:
This is rather inconsequential for `N = 2`, but it would be "faster" to check for the key's existence in the hash rather than to create and then iterate over a list of all keys.